src/Html.php

```diff
@@ -1181,24 +1301,46 @@ public static function checkboxList(string $name, $selection = null, array $item
      *
      * See {@see renderTagAttributes()} for details on how attributes are being rendered.
      *
+     * @psalm-param iterable<array-key, string>|string|\Stringable|int|float|bool|null $selection
+     * @psalm-param array<array-key, string> $items
+     * @psalm-param InputHtmlOptions&array{
+     *   item?: Closure(int, string, string, bool, mixed):string|null,
+     *   itemOptions?: HtmlOptions|null,
+     *   encode?: bool,
+     *   separator?: string|null,
+     *   tag?: string|null,
+     *   unselect?: string|int|float|\Stringable|bool|null,
+     * }|array<empty, empty> $options
+     *
      * @throws JsonException
      *
      * @return string the generated radio button list
      */
     public static function radioList(string $name, $selection = null, array $items = [], array $options = []): string
     {
         if (is_iterable($selection)) {
-            $selection = array_map('strval', (array)$selection);
+            $selection = array_map('strval', is_array($selection) ? $selection : iterator_to_array($selection));
```

Comment:
Originally it meant that if a single value is selected then you can specify it without wrapping it into array explicitly. Now it changed its meaning.